src/router.ts

```diff
@@ -47,16 +60,46 @@ router.add('GET', '/rtv/metadata', async (request) => {
   );
 });
 
+/**
+ * Handles amp-geo requests.
+ * @param rtv - RTV number to use.
+ * @param path - path to the requested amp-geo file, must start with `/`.
+ * @param cf - incoming Cloudflare properties.
+ * @returns a dynamically injected amp-geo response, possibly from cache.
+ */
+async function ampGeoRequest(
+  rtv: string,
+  path: string,
+  cf: IncomingCloudflarePropertiesExtended
+): Promise<Response> {
+  const url = getAmpFileUrl(rtv, path);
+  const cacheKey = `${cf.country ?? ''};${cf.regionCode ?? ''}`;
+
+  const cachedResponse = await injectorsCache.getCacheFor(url, cacheKey, cf);
```

Comment:
Nit: Given that we're checking cache here, I'm not sure the `index.ts` file is the right level for putting into the cache. Seems like it'd have a lot less indirection by putting an out-of-band call right at the bottom of `ampGeoRequest`.

Reply:
We need at `event` object (to call `event.waitUntil`), which isn't passed to the `router` instance's handler functions (`event.request` is). I agree that this isn't exactly idea, and would be happy to figure out a better structure for this

Reply:
Seems the Request object has an `extend` method which is just a bound `event.waitUntil`: https://github.com/lukeed/worktop/blob/e3a44ba8cd34d7fa6849f98d4fb8dae37afdb404/src/request.ts#L12

Reply:
Oh sweet! Nice catch :D
I think with this it's possible to complete get rid of the internal header, I'll give it a stab

Reply:
Done 022b831, wdyt?